raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -44,7 +44,7 @@
  *    Follower: After discovering a leader with an equal or larger epoch
  *
  * Unattached transitions to:
- *    Unattached: After learning of a new election with a higher epoch or after voting
+ *    Unattached: After learning of a new election with a higher epoch or after giving a binding vote
```

Comment:
This applies to the "Unattached transitions" and "Voted transitions" section.

Should we merge this two and the wording since in a previous PR we merge these two states.

Reply:
looks like I changed it in the [second PR ](https://github.com/apache/kafka/pull/18041/files#diff-095b9e60b0227d41f24b49923afddbe813d5c5754cd0ef4801741f1d275319b6L51-R51) when I added the new Prospective state. I'll fix it here too